integration-tests/validation/src/main/java/com/android/designcompose/testapp/validation/examples/StaticExamples.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ package com.android.designcompose.testapp.validation.examples
+
+ import androidx.compose.runtime.Composable
+
+ import com.android.designcompose.DesignDocOverride
+ import com.android.designcompose.annotation.Design
+ import com.android.designcompose.annotation.DesignComponent
+ import com.android.designcompose.annotation.DesignDoc
+ 
+ // Layout "stage"
+@DesignDoc(id = "Gv63fYTzpeH2ZtxP4go31E")
+private interface LayoutStage {
+    @DesignComponent(node = "#stage") fun MainFrame()
+}
+@Composable
+private fun LayoutStage() {
+    LayoutStageDoc.MainFrame()
+}
+
+// Layout "stuff"
+@DesignDoc(id = "Gv63fYTzpeH2ZtxP4go31E")
+private interface LayoutStageStuff {
+    @DesignComponent(node = "#stage-stuff") fun MainFrame()
+}
+@Composable
+private fun LayoutStageStuff() {
+    LayoutStageStuffDoc.MainFrame()
+}
+
+// Blend mode test
+@DesignDoc(id = "ZqX5i5g6inv9tANIwMMXUV")
+private interface BlendModeStage {
+    @DesignComponent(node = "#stage") fun MainFrame()
+}
+@Composable
+private fun BlendModeStage() {
+    BlendModeStageDoc.MainFrame()
+}
+
+// Absolute Layout Alignment Test node
+@DesignDoc(id = "JIjE9oKQbq8ipi66ab5UaK")
+private interface AbsoluteLayoutAlignmentTest {
+    @DesignComponent(node = "#Test") fun MainFrame()
+}
+@Composable
+private fun AbsoluteLayoutAlignmentTest() {
+    AbsoluteLayoutAlignmentTestDoc.MainFrame()
+}
+ 
+
+ // To run these tests in a local development env, run `./gradlew recordRoborazziDebug`.file
+ // Test outputs should be a series of images at:
+ // `integration-tests/validation/src/testDebug/roborazzi/RenderStaticExamples/*.png`
+ val STATIC_EXAMPLES: ArrayList<Triple<String, @Composable () -> Unit, String?>> =
+     arrayListOf(
+         // TODO: There's a problem with code generation so that I have to keep these tets in place or
+         // nodes can't be found for subsequent tests.
+         Triple("Layout Tests", { LayoutTests() }, LayoutTestsDoc.javaClass.name),
+         Triple("Alignment", { AlignmentTest() }, AlignmentTestDoc.javaClass.name),
+         Triple("Blend Modes", { BlendModeTest() }, BlendModeTestDoc.javaClass.name),
```

Comment:
@timothyfroehlich Tim, This code currently runs and provides images. But only if I bring in these tests from `AllExamples.kt`.

I believe I'm not correctly hooking the tests defined in this module into the code generation system, requiring these external tests to be brought in.
Any idea what hook I'm missing here?

Reply:
Check out the commit that I just pushed. The `@DesignDoc` annotation creates a composables named after each `@DesignComponent` in the `@DesignDoc` (such as `MainFrame()`). We can use these directly in the tests, we don't need to create a separate composable that contains them.

Also, you need to fetch the Figma files before running screenshot tests. You can run `./gradlew val:fAUFF` (short for validation-app:fetchAndUpdateFigmaFiles) to fetch them all. That task is a fairly dumb one and I don't know if there's a way to filter it to only fetch some, so I'd recommend commenting out the `testRender` test in `RenderAllExamples` so you don't have to wait for all of the Validation examples to re-fetch.

Reply:
I've updated to include your commit. Thanks Tim.
Was able to include a fair number of additional tests, but some (commented out) are still failing, sometimes at the "fAUFF" pass, sometimes in the "recordRoborazziDebug" stage.
I'll look at those next week.